File: 프로그래머스/1/388351. 유연근무제/유연근무제.cpp
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

static int Work(vector<int>& schedules, vector<vector<int>>& timelogs, queue<int> day) {
    int result = 0;

    for (int i = 0; i < schedules.size(); i++) {
        queue<int> days = day; 
        bool isEligible = true; 
        int schedule = schedules[i] + 10;
        if(schedule % 100 >= 60) schedule += 40;

        for (int j = 0; j < 7; j++) {
            int current_day = days.front();
            days.pop();

            if (current_day == 6 || current_day == 7) {
                continue; 
            }

            if (timelogs[i][j] > schedule) {
                isEligible = false; 
                break;
            }
        }

        if (isEligible) {
            result++;
        }
    }
    return result;
}

int solution(vector<int> schedules, vector<vector<int>> timelogs, int startday) {
    int answer = 0;
    queue<int> day;

    for (int i = 0; i < 7; i++) {
        day.push((startday + i - 1) % 7 + 1);
    }

    answer = Work(schedules, timelogs, day);

    return answer;
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.js
function solution(schedules, timelogs, startday) {
    var answer = 0;

    function getScheduleTime(num) {
        const hour = Math.floor(num / 100);
        const minute = Math.floor(num % 100);

        let total = (hour * 60) + minute + 10;

        const nhour = Math.floor(total / 60);
        const nminute = Math.floor(total % 60);

        return nhour * 100 + nminute;
    }

    for (let i = 0; i < schedules.length; i++) {
        const s = getScheduleTime(schedules[i]);

        let d = startday - 1;
        let flag = true;
        for (let j = 0; j < 7; j++) {
            d++;
            if (d % 7 === 6 || d % 7 === 0) continue;
            if (s < timelogs[i][j]) {
                flag = false;
                break;
            }
        }

        if (flag) answer++;
    }

    return answer;
}